### ISSUE TITLE
bug: fit_batches training loop

### ISSUE THREAD

**Issue by Daniel-Dodd** [thomaspinder/GPJax#96]
# Bug Report

**GPJax version: 0.4.10, 0.4.9**

**Current behavior:**

There is an issue regarding `fit_batches`. I am still diagnosing this, but I believe the TensorFlow dataset iterator (for mini-bathing) has issues within the `lax.scan`. I noticed this because I am unable to reproduce previous experimental results (from Easter) with older GPJax code. Furthermore, plotting the ELBO results in a smooth curve, which is not expected during stochastic optimisation, and the final model predictions are poor. Each training step appears to cache and reuse the same training batch of data. Increasing the batch size appears to improve performance.

**Expected behavior:**

Running the model on an "old" GPJax training abstraction, with the same model, the same gradient optimiser settings, same number of loop iterations, same random numbers, gives a rough jagged ELBO training curve, as expected. And the fit is excellent, reproducing my previous results.

**Steps to reproduce:**

I'll provide an "old" training loop and or notebook example in a future comment.

**Comment by thomaspinder** [comment]
Resolved in #99
